src/classes/PS_IntegrationService_TEST.cls

```diff
@@ -35,16 +35,112 @@
 * @description Test class for the PS_IntegrationService class
 */
 
-@isTest
+@IsTest
 private class PS_IntegrationService_TEST {
-    
+
+    private static Map<String, List<Map<String, String>>> configParams;
+    private static Map<String, PS_IntegrationService.Config> makanaConfigParams;
+    private static Map<String, PS_IntegrationService.Config> paymentConfigParams;
+    private static final String API_KEY = 'apiKey';
+    private static final String PAYMENTS_APPLICATION_TYPE = 'payments';
+    private static final String MAKANA_APPLICATION_TYPE = 'makana';
+
+    static {
+        buildConfigurationParams(); // Build Test Configuration Parameters
+    }
+
     /*******************************************************************************************************
-     * @description tests storing of the secrets we receive from the Payment Services
-     * Integrations package
+     * @description Test for storing the secrets we receive from the Payment Services
+     * Integrations package with a valid API Key
      */
-    @isTest
-    private static void setConfigurationTest() {
-        // TODO This method can be implemented after the setConfiguration method is implemented
-         
+    @IsTest
+    private static void shouldSetConfigurationWithPaymentsAPIKey() {
+        Test.startTest();
+
+        PS_IntegrationService.setConfiguration(configParams);
+
+        List<Payment_Services_Configuration__c> configurations = [
+                SELECT Key__c, Service_Key__c, Value__c, Is_Secret__c, Service__c
+                FROM Payment_Services_Configuration__c
+        ];
+        Test.stopTest();
+        System.assert(!configurations.isEmpty());
+        for (Payment_Services_Configuration__c config : configurations) {
+         if (config.Service__c == PAYMENTS_APPLICATION_TYPE) {
```

Comment:
There's some mis-aligned indents through this class, and this is one of them - looks like these are all one-space instead of four